finance_dl/amazon.py

```diff
@@ -185,10 +201,19 @@ def retrieve_all_order_groups():
                     break
 
         if regular:
-            (orders_link,), = self.wait_and_return(
-                lambda: self.find_elements_by_descendant_text_match('. = "Orders"', 'a', only_displayed=True)
-            )
-            scrape_lib.retry(lambda: self.click(orders_link), retry_delay=2)
+            if self.amazon_domain == Domain.CO_UK:
+                # on co.uk, orders link is hidden behind the menu, hence not directly clickable
+                (orders_link,), = self.wait_and_return(
```

Comment:
I think this logic would work for both domains, right?  In which case we could simplify the code